code/modules/mob/living/simple_animal/abnormality/teth/punishing_bird.dm

```diff
@@ -227,7 +227,7 @@
 
 /mob/living/simple_animal/hostile/abnormality/punishing_bird/proc/kill_bird()
 	if(!(status_flags & GODMODE) && !target && icon_state != "pbird_red")
-		QDEL_NULL(src)
+		death() // QDel BREAKS anything that requires him to DIE
```

Comment:
There's a good reason why we look for Destroy() proc instead of death() almost everywhere.